base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -2251,27 +2222,41 @@ fn find_orphan_descendant_tips_of<T: BlockchainBackend>(
             target: LOG_TARGET,
             "Found new orphan tip {} ({})",
             &prev_chain_header.height(),
-            &prev_chain_header.hash().to_hex()
+            &prev_chain_header.hash(),
         );
         return Ok(vec![prev_chain_header]);
     }
 
+    debug!(
+        target: LOG_TARGET,
+        "Found {} children of orphan {} ({})",
+        children.len(),
+        &prev_chain_header.height(),
+        &prev_chain_header.hash()
+    );
+
+    let mut current_branch_prev_timestamps = prev_timestamps.clone();
+
     let mut res = vec![];
     for child in children {
         debug!(
             target: LOG_TARGET,
             "Validating header #{} ({}), descendant of #{} ({})",
             child.header.height,
-            child.hash().to_hex(),
+            child.hash(),
             prev_chain_header.height(),
-            prev_chain_header.hash().to_hex()
+            prev_chain_header.hash(),
         );
+        // Append the child timestamp - a RollingVec ensures that the number of timestamps can never be more than the
+        // median timestamp window size.
+        current_branch_prev_timestamps.push(child.header.timestamp);
```

Comment:
this should only happen after validation, not before

Reply:
thanks you are correct :)